src/js/components/tweetsContainer.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import TweetBlock from './tweetBlock';
+import {promiseTimeline, promiseTimelineSelf} from '../services/timelineService';
+
+export const DEFAULT_ERROR_MESSAGE = 'Something went wrong, please contact systems administrator.';
+
+export default class TweetsContainer extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            tweets: [], // tweets retrieved 
+            isLoaded: false, // flags if request was loaded or not
+            error: null, // error message
+        };
+         
+        if(this.props.type in this.TYPES){ // change type to specified type, if specified type exists
+            this.serviceMethod = this.TYPES[this.props.type]
+        } else {
+            this.serviceMethod = this.TYPES['default']; // set type to default
+        }
+    }
+
+    get TYPES(){
+        return {
+            default: () => promiseTimeline(),
+            self: () => promiseTimelineSelf().then(
+                (response) => {
+                    if(response?.data?.length == 0){ // if no tweets, set error message to special error message
+                        this.errorMessage = 'No tweets are available, post a tweet!';
+                        throw new Error('No tweets');
+                    } else {
+                        response.data.map((tweet) => {  // remove twitterHandle from tweets that are passed to TweetBlock
+                            delete tweet.user.twitterHandle;
```

Comment:
why the deletion? If someone was to pass data in, you may mistakenly modify the incoming data.  Also  reusable components

Reply:
I think I understand. Would defining a new variable instead work? For example: 
`let resp = {
                            data: response.data.map((tweet) => {  TweetBlock
                                delete tweet.user.twitterHandle;
                                return tweet;
                            })
                        }
                        return resp;`
Also, could you clarify what you meant by 'Also reusable components'?

Reply:
yeah it's a difference between data vs component/rendering

Reply:
Updated